### PR TITLE
fix javaasist collision

### PR DESCRIPTION
**Pull request by jcabrerizo**
Brookyn build is failing when you try to package as RMP: https://ci-builds.apache.org/job/Brooklyn/job/brooklyn/job/master/5/console

The error is
```
Failed while enforcing releasability the error(s) are [
Dependency convergence error for org.javassist:javassist:3.24.0-GA paths to dependency are:
+-org.apache.brooklyn:rpm-packaging:1.1.0-SNAPSHOT
  +-org.apache.brooklyn:apache-brooklyn:1.1.0-SNAPSHOT
    +-org.apache.karaf.features:enterprise:4.2.8
      +-org.hibernate:hibernate-osgi:5.4.8.Final
        +-org.hibernate:hibernate-core:5.4.8.Final
          +-org.javassist:javassist:3.24.0-GA
and
+-org.apache.brooklyn:rpm-packaging:1.1.0-SNAPSHOT
  +-org.apache.brooklyn:apache-brooklyn:1.1.0-SNAPSHOT
    +-org.apache.brooklyn:brooklyn-features:1.1.0-SNAPSHOT
      +-org.apache.brooklyn:brooklyn-rest-resources:1.1.0-SNAPSHOT
        +-org.reflections:reflections:0.9.10
          +-org.javassist:javassist:3.19.0-GA
```

### CODE REVIEW

File: karaf/apache-brooklyn/pom.xml
Comment (geomacy) [inline karaf/apache-brooklyn/pom.xml:66]: Just for info, what is the collision here?  Perhaps you could add a line or two in the PR description.

Comment (jcabrerizo) [reply karaf/apache-brooklyn/pom.xml:66]: Hi @geomacy,
Brookyn build is failing when you try to package as RMP. You can see in the last Master try-of-build: https://ci-builds.apache.org/job/Brooklyn/job/brooklyn/job/master/5/console

The error is
```
Failed while enforcing releasability the error(s) are [
Dependency convergence error for org.javassist:javassist:3.24.0-GA paths to dependency are:
+-org.apache.brooklyn:rpm-packaging:1.1.0-SNAPSHOT
  +-org.apache.brooklyn:apache-brooklyn:1.1.0-SNAPSHOT
    +-org.apache.karaf.features:enterprise:4.2.8
      +-org.hibernate:hibernate-osgi:5.4.8.Final
        +-org.hibernate:hibernate-core:5.4.8.Final
          +-org.javassist:javassist:3.24.0-GA
and
+-org.apache.brooklyn:rpm-packaging:1.1.0-SNAPSHOT
  +-org.apache.brooklyn:apache-brooklyn:1.1.0-SNAPSHOT
    +-org.apache.brooklyn:brooklyn-features:1.1.0-SNAPSHOT
      +-org.apache.brooklyn:brooklyn-rest-resources:1.1.0-SNAPSHOT
        +-org.reflections:reflections:0.9.10
          +-org.javassist:javassist:3.19.0-GA
```
This PR solved on my local the issue, once it's approved we could see if it also fix it on Jenkins.

Thanks for pointing the lack of background, probably will be helpful in the future if someone wants we are excluding this.

Comment (geomacy) [reply karaf/apache-brooklyn/pom.xml:66]: Thanks Juan. LGTM.

**Review by geomacy — APPROVED** [review]
LGTM

**Pull request merged**
